### PR TITLE
pin versions

### LINKED ISSUE #5: DataCloneError: WebAssembly.Memory object could not be cloned.
I'm very new to Rust so please assume the worst.

**Describe the bug**
I'm getting `WebAssembly.Memory object could not be cloned` in Firefox and Chrome on the latest Main and [the latest human commit](https://github.com/tweag/rust-wasm-threads/commit/d8b62793108130b0fe86e9b9d43098eb57b164f1) for `shared-memory`. 

`separate-memory` and `separate-memory-transfer` work as expected, presumably because they don't use [`wasm_bindgen::memory()`](https://github.com/tweag/rust-wasm-threads/blob/5921917884800509fdfe6156460d09a86ea7be31/shared-memory/src/lib.rs#L60) which I assume cannot be cloned.

**To Reproduce**

I pinned wasm-bindgen to `"=0.2.81"` [here](https://github.com/tweag/rust-wasm-threads/blob/5921917884800509fdfe6156460d09a86ea7be31/separate-memory-transfer/Cargo.toml#L12) (for both the latest Main and human commit) because otherwise it kept trying to use 0.2.83 - I blame [the workspace lock file](https://github.com/tweag/rust-wasm-threads/blob/5921917884800509fdfe6156460d09a86ea7be31/Cargo.lock#L157).

```
error:

it looks like the Rust project used to create this wasm file was linked against
version of wasm-bindgen that uses a different bindgen format than this binary:

  rust wasm file schema version: 0.2.83
     this binary schema version: 0.2.81
```

(I've no idea how Nix (flakes) works.) I then ran

```
cargo install --version 0.2.81 wasm-bindgen-cli --force
cd shared-memory
cargo build --release
wasm-bindgen ../target/wasm32-unknown-unknown/release/shared_memory.wasm --target=no-modules --out-dir=pkg
python ../server.py`
```

which resulted in the stacktrace at the bottom of this issue.

**Expected behavior**
No error.

**Environment**
 - OS name + version: Ubuntu 20.04.4 LTS (really using WSL though)
 - Chrome 119.0.0.0
 - Firefox 119.0

**Additional context**

I found this potentially relevant [SO question](https://stackoverflow.com/q/59705741), but reading the generated `pkg/shared_memory.js` file doesn't reveal any `new WebAssembly.Memory` calls.

Complete stacktrace:
```
panicked at shared-memory/src/lib.rs:41:6:
failed to start worker 1: JsValue(DataCloneError: WebAssembly.Memory object could not be cloned.
getImports/imports.wbg.__wbg_postMessage_8a8f33d997e17e7b/<@http://localhost:8000/pkg/shared_memory.js:282:25
handleError@http://localhost:8000/pkg/shared_memory.js:201:18
getImports/imports.wbg.__wbg_postMessage_8a8f33d997e17e7b@http://localhost:8000/pkg/shared_memory.js:281:74
web_sys::features::gen_Worker::Worker::post_message::hfc0de0597810731c@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[138]:0x10d01
run@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[22]:0x5938
__exports.run@http://localhost:8000/pkg/shared_memory.js:189:10
promise callback*@http://localhost:8000/index.js:2:45
)

Stack:

getImports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://localhost:8000/pkg/shared_memory.js:256:21
console_error_panic_hook::hook::h2d269cc0da8a6d49@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[50]:0xbcee
core::ops::function::Fn::call::h4825d0fc11fc7cfd@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[246]:0x1196f
std::panicking::rust_panic_with_hook::h3aa054d35a0817d7@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[89]:0xf3ba
std::panicking::begin_panic_handler::{{closure}}::h2f73e4cf6cd6319a@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[105]:0x10075
std::sys_common::backtrace::__rust_end_short_backtrace::h98ac61a6abbff7e9@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[241]:0x11947
rust_begin_unwind@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[123]:0x1088c
core::panicking::panic_fmt::h3e1dd3d08288569e@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[145]:0x10ee4
core::result::unwrap_failed::h8b3db0f11171b57b@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[113]:0x104c7
run@http://localhost:8000/pkg/shared_memory_bg.wasm:wasm-function[22]:0x5c41
__exports.run@http://localhost:8000/pkg/shared_memory.js:189:10
promise callback*@http://localhost:8000/index.js:2:45
```

### PR DESCRIPTION
**Pull request by AlexErrant**
Closes https://github.com/tweag/rust-wasm-threads/issues/5

Here's a starting point :)

I think nightly is only needed for shared-memory. I went with `nightly-2023-06-15` because the more recent ones broke for reasons that are beyond me so I blindly went with [this answer](https://substrate.stackexchange.com/a/9069).

### CODE REVIEW

File: separate-memory-transfer/README.md [hunk 1]
```diff
@@ -4,10 +4,11 @@ An example in which we spawn a worker and transfer a buffer back and forth betwe
 and the worker.
 
 To run this example, `cd` into this directory and then:
-```rust
-> cargo build --release
-> wasm-bindgen ../target/wasm32-unknown-unknown/release/separate_memory_transfer.wasm --target=no-modules --out-dir=pkg
-> python ../server.py
+```sh
+cargo install --version 0.2.81 wasm-bindgen-cli --force
```
Comment (jneem) [inline separate-memory-transfer/README.md]:
```suggestion
cargo install --version 0.2.81 wasm-bindgen-cli --force # if you aren't getting it from the nix flake
```

(For people using nix, the `flake.nix`/`flake.lock` files already provide the correct version of `wasm-bindgen-cli`)

File: separate-memory/README.md [hunk 2]
```diff
@@ -4,10 +4,11 @@ An example in which we spawn a worker and copy a buffer back and forth between t
 and the worker.
 
 To run this example, `cd` into this directory and then:
-```rust
-> cargo build --release
-> wasm-bindgen ../target/wasm32-unknown-unknown/release/separate_memory.wasm --target=no-modules --out-dir=pkg
-> python ../server.py
+```sh
+cargo install --version 0.2.81 wasm-bindgen-cli --force
```
Comment (jneem) [inline separate-memory/README.md]: here too

File: shared-memory/README.md [hunk 3]
```diff
@@ -4,10 +4,11 @@ A simple example in which we spawn a worker in the same address space as its par
 send a buffer back and forth using a rust channel.
 
 To run this example, `cd` into this directory and then:
-```rust
-> cargo build --release
-> wasm-bindgen ../target/wasm32-unknown-unknown/release/shared_memory.wasm --target=no-modules --out-dir=pkg
-> python ../server.py
+```sh
+cargo install --version 0.2.81 wasm-bindgen-cli --force
```
Comment (jneem) [inline shared-memory/README.md]: and here

**Review by jneem — COMMENTED** [review]
Thanks for doing the work for me :smile: 

Just one (or maybe three) small comment(s) to not confuse people that are getting dependencies from the nix flake.

**Comment by AlexErrant** [comment]
Rubbing my two braincells together and clicking a few more links led me to https://github.com/rust-lang/rust/issues/113152 so I deleted all the Cargo.lock files and recreated the workspace lockfile - now the latest nightly works.

**Comment by jneem** [comment]
Ah, I originally started with three separate workspaces and then when I merged it into one I guess I forgot that there should only be the top-level lockfile...

Thanks!

**Pull request merged**